File: medplum-java/medplum-model/src/main/java/com/fhir/ActivityDefinition_Participant.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableActivityDefinition_Participant.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ActivityDefinition_Participant {
  Optional<code> type();

  Optional<String> id();

  Optional<CodeableConcept> role();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  static ImmutableActivityDefinition_Participant.Builder builder() {
    return ImmutableActivityDefinition_Participant.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/ActivityDefinition_Participant.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";

export interface ActivityDefinition_Participant {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: code | null | undefined;
    role: CodeableConcept | null | undefined;
}
